Clarify `Texture Atlas rework` migration guide (#1028)
bevyengine/bevy-website: content/learn/migration-guides/0.12-to-0.13.md

`SpriteSheetBundle` now uses a `Sprite` instead of a `TextureAtlasSprite` component and a `TextureAtlas` component instead of a `Handle<TextureAtlaslayout>`.

> `DynamicTextureAtlasBuilder::add_texture` takes an addition `&Handle<Image>` parameter.

Reviewer: ```suggestion
`DynamicTextureAtlasBuilder::add_texture` takes an additional `&Handle<Image>` parameter.
```

`TextureAtlasLayout::from_grid` no longer takes a `Handle<Image>` parameter.

> `TextureAtlasBuilder::finish` now returns a `Result<(TextureAtlasLayout, Handle<Image>), _>`.

Reviewer: This doesn't seem right

> `UiAtlasImage` was removed. The atlas bundle is now identical to `ImageBundle` with an additional `TextureAtlas`.

Reviewer: ```suggestion
`UiAtlasImage` was removed. Use `ImageBundle` and add an additional `TextureAtlas` component.
```

Author: This is confusing. https://docs.rs/bevy/latest/bevy/prelude/struct.AtlasImageBundle.html does exist; why shouldn't this be used like the changelog suggests?

Reviewer: Oops, this is nonsense from me.

Reviewer: I think that it was `UiTextureAtlasImage` that was removed though.
